.github/actions/release-archive/action.yml

```diff
@@ -0,0 +1,47 @@
+name: Upload release archive
+description: Bundles an archive of artifacts and uploads to a GitHub release. The artifacts must already have been built into the folder "install".
+inputs:
+  os:
+    description: The OS runner name being built, used in the archive name
+    required: true
+  upload_to_release:
+    description: Whether to upload the archives to the release
+runs:
+  using: composite
+  steps:
+    - name: archive
+      id: archive
+      shell: bash
+      run: |
+        OSNAME=$(echo ${{ inputs.os }} | sed 's/-latest//')
+        VERSION=${{ github.ref_name }}
```

Comment:
Note this significant change, `ref_name` is still the tag for release build but the branch name for normal one